definitions/default.rb

```diff
@@ -33,6 +33,13 @@ def multipackage_definition_impl(params)
 
     t = begin
           # non-delayed eager accumulators like this cannot use recursive search
+          #
+          # If we are being called from within a custom resource and...
```

Comment:
should probably be "If we are being called from within a custom resource and use recursive search and find a resource in an outer run context, then we always get the use case wrong because:"

which is a hell of a mouthful, but otherwise you're leaving out details...